src/05-derivatives/02-common-data-types.md

```diff
@@ -30,13 +30,29 @@ Note that even though `space` and `desc` are
 optional at least one of them needs to be defined to avoid name conflict with
 the raw file.
 
+**Sampling**. When two or more instances of a given derivative are provided
+with resolution (or surface sampling density) being the only difference
+between them, then the `res` (for *resolution* of regularly sampled N-D data)
+and/or `den` (for *density* of non-parametric surfaces) SHOULD be used to avoid
+name conflicts.
+Note that only files combining both regularly sampled (e.g., gridded) and surface
+sampled data (and their downstream derivatives) are allowed to present both `res`
+and `den` keywords simultaneously.
+
+!!! tip "Recommendation"
```

Comment:
I think this is better phrased using RFC 2119 terms for consistency with documentation elsewhere and [overall common principles](https://bids-specification.readthedocs.io/en/stable/02-common-principles.html).

Reply:
Note that `RECOMMENDED` is an option there.

Reply:
I'll revise and define this as OPTIONAL